contracts/contracts/strategies/plume/RoosterAMOStrategy.sol

```diff
@@ -0,0 +1,1181 @@
+    /// the pre-configured pool's liquidity ticker
+    uint256 public allowedWethShareEnd;
+    /// @dev reserved for inheritance
+    int256[46] private __reserved;
+
+    /***************************************
+          Constants, structs and events
+    ****************************************/
+
+    /// @notice The address of the Wrapped ETH (WETH) token contract
+    address public immutable WETH;
+    /// @notice The address of the OETHp token contract
+    address public immutable OETHp;
+    /// @notice the underlying AMO Maverick (Rooster) pool
+    IMaverickV2Pool public immutable mPool;
+    /// @notice the Liquidity manager used to add liquidity to the pool
+    IMaverickV2LiquidityManager public immutable liquidityManager;
+    /// @notice the Maverick V2 poolLens
+    ///
+    /// @dev only used to provider the pool's current sqrtPrice
+    IMaverickV2PoolLens public immutable poolLens;
+    /// @notice the Maverick V2 position
+    ///
+    /// @dev provides details of the NFT LP position and offers functions to
+    /// remove the liquidity.
+    IMaverickV2Position public immutable maverickPosition;
+    /// @notice the Maverick Quoter
+    IMaverickV2Quoter public immutable quoter;
+    /// @notice the Maverick Voting Distributor
+    IVotingDistributor public immutable votingDistributor;
+    /// @notice the Maverick Pool Distributor
+    IPoolDistributor public immutable poolDistributor;
+
+    /// @notice sqrtPriceTickLower
+    /// @dev tick lower represents the lower price of OETHp priced in WETH. Meaning the pool
+    /// offers more than 1 OETHp for 1 WETH. In other terms to get 1 OETHp the swap needs to offer 0.9999 WETH
+    /// this is where purchasing OETHp with WETH within the liquidity position is the cheapest.
+    ///
+    ///            _____________________
+    ///            |      |            |
+    ///            | WETH |    OETHp   |
+    ///            |      |            |
+    ///            |      |            |
+    ///  --------- * ---- * ---------- * ---------
+    ///               currentPrice
+    ///                          sqrtPriceHigher-(1:1 parity)
+    ///      sqrtPriceLower
+    ///
+    ///
+    /// Price is defined as price of token1 in terms of token0. (token1 / token0)
+    /// @notice sqrtPriceTickLower - OETH is priced 0.9999 WETH
+    uint256 public immutable sqrtPriceTickLower;
+    /// @notice sqrtPriceTickHigher
+    /// @dev tick higher represents 1:1 price parity of WETH to OETHp
+    uint256 public immutable sqrtPriceTickHigher;
+    /// @dev price at parity (in OETHp this is equal to sqrtPriceTickHigher)
+    uint256 public immutable sqrtPriceAtParity;
+    /// @notice The tick where the strategy deploys the liquidity to
+    int32 public constant TICK_NUMBER = -1;
+    /// @notice Minimum liquidity required to continue with the action
+    /// e.g. deposit, add liquidity, burn OETHp
+    uint256 public constant MIN_BALANCE_THRESHOLD = 1e12;
```

Comment:
A "Min balance threshold" sounds like we want to keep this much money in the contract. Perhaps it should be `ACTION_THRESHOLD` to show it controls if something happens or not?

Reply:
better thanks: https://github.com/OriginProtocol/origin-dollar/pull/2499/commits/12350c8f57307f7949edc90cae0479ddccb9964c